tools/eval/datasets/pp_humanseg.py

```diff
@@ -0,0 +1,209 @@
+                output_image,
+                expected_image,
+                self.num_classes)
+            intersect_area_all = intersect_area_all + intersect_area
+            pred_area_all = pred_area_all + pred_area
+            label_area_all = label_area_all + label_area
+
+
+            
+        self.class_iou, self.miou = self.mean_iou(intersect_area_all, pred_area_all,
+                                           label_area_all)
+        self.class_acc, self.acc = self.accuracy(intersect_area_all, pred_area_all)
+    
+
+    def get_results(self) :
+        """
+        Get evaluation results
+        Returns :
+            miou (float) : mean iou
+            class_miou (list) : iou on all classes
+            acc (float) : mean accuracy
+            class_acc (list) : accuracy on all classes
+        """
+        return self.miou, self.class_miou, self.acc, self.class_acc
+    
+
+    def print_result(self) : 
+        """
+        Print evaluation results
+        """
+        print("Mean IoU : ", self.miou)
+        print("Class IoU : ", self.class_iou)
+        print("Mean Accuracy : ", self.acc)
+        print("Class Accuracy : ", self.class_acc)
+
+    
+    def one_hot(self, arr, max_size) :
+        return np.eye(max_size)[arr]
+
+
+    def calculate_area(self,pred, label, num_classes, ignore_index=255):
+        """
+        Calculate intersect, prediction and label area
+        Args:
+            pred (Tensor): The prediction by model.
+            label (Tensor): The ground truth of image.
+            num_classes (int): The unique number of target classes.
+            ignore_index (int): Specifies a target value that is ignored. Default: 255.
+        Returns:
+            Tensor: The intersection area of prediction and the ground on all class.
+            Tensor: The prediction area on all class.
+            Tensor: The ground truth area on all class
+        """
+        
+        # Delete ignore_index
+        mask = label != ignore_index
+        pred = pred + 1
+        label = label + 1
+        pred = pred * mask
+        label = label * mask
+
+
+        pred = self.one_hot(pred, num_classes + 1)
+        label = self.one_hot(label, num_classes + 1)
+
+        pred = pred[:, :, :, 1:]
+        label = label[:, :, :, 1:]
+
+        pred_area = []
+        label_area = []
+        intersect_area = []
+
+        #iterate over all classes and calculate their respective areas
+        for i in range(num_classes):
+            pred_i = pred[:, :, :, i]
+            label_i = label[:, :, :, i]
+            pred_area_i = np.sum(pred_i)
+            label_area_i = np.sum(label_i)
+            intersect_area_i = np.sum(pred_i * label_i)
+            pred_area.append(pred_area_i)
+            label_area.append(label_area_i)
+            intersect_area.append(intersect_area_i)
+        
+        return intersect_area, pred_area, label_area
+    
+
+    def mean_iou(self,intersect_area, pred_area, label_area):
```

Comment:
Could you provide some test data for validating the correctness of this function? Something like test cases with given input and expected output.

Reply:
please refer above